codec-http2/src/main/java/io/netty5/handler/codec/http2/DefaultHttp2FrameWriter.java

```diff
@@ -219,32 +225,31 @@ public Future<Void> writeData(ChannelHandlerContext ctx, int streamId, ByteBuf d
                     // Write the payload.
                     if (data != null) { // Make sure Data is not null
                         if (remainingData == 0) {
-                            ByteBuf lastFrame = data.readSlice(frameDataBytes);
+                            Buffer lastFrame = data.readSplit(frameDataBytes);
+                            data.close();
                             data = null;
                             ctx.write(lastFrame).cascadeTo(promiseAggregator.newPromise());
                         } else {
-                            ctx.write(data.readRetainedSlice(frameDataBytes))
+                            ctx.write(data.readSplit(frameDataBytes))
                                     .cascadeTo(promiseAggregator.newPromise());
                         }
                     }
                     // Write the frame padding.
                     if (paddingBytes(framePaddingBytes) > 0) {
-                        ctx.write(ZERO_BUFFER.slice(0, paddingBytes(framePaddingBytes)))
+                        ctx.write(ZERO_BUFFER.copy(0, paddingBytes(framePaddingBytes), true))
```

Comment:
To get a buffer with independent life time. To be safer, we can also use a const buffer supplier here.

Reply:
Turns out we can't use a const buffer supplier because those can't be resized like we need to here.
The read-only shenanigans means we effectively get a retained slice here anyway.